### PR TITLE
1:1 OSRS Pathing - Ynneh

### PR DESCRIPTION
**Pull request by tobywisener**
Pathing completed by Ynneh, fixed now needs tested and needs tidied up

### CODE REVIEW

File: ElvargClient/src/main/java/com/runescape/io/PacketConstants.java [hunk 1]
```diff
@@ -132,7 +132,7 @@ public final class PacketConstants {
             -1, 0, 0, 0, 13, 0, 0, 1, 0, 0, // 130
             3, 10, 2, 0, 0, 0, 0, 14, 0, 0, //140
             0, 4, 5, 3, 0, 0, 3, 0, 0, 0, //150
-            4, 5, 0, 0, 2, 0, 6, 0, 0, 0, //160
```
Comment (tobywisener) [inline ElvargClient/src/main/java/com/runescape/io/PacketConstants.java:135]: This fixes the cooking packet that crashes gameplay

File: ElvargServer/src/main/java/com/elvarg/game/content/combat/Combat.java [hunk 2]
```diff
@@ -84,12 +87,32 @@ public void performNewAttack(boolean instant) {
...
+
+            int attackDistance = target.getMovementQueue().isMoving() && method.type() == CombatType.MELEE ? 2 : method.attackDistance(character);
+
+            int distanceFromTarget = character.getLocation().getDistance(target.getLocation());
+
+            boolean canReach = RegionManager.canProjectileAttack(character, target);
+
+            boolean needsPath = distanceFromTarget > attackDistance || !canReach;
+
+            System.err.println("MaxAttackDistance=" + attackDistance + ", CurrentDistance=" + distanceFromTarget + ", path=" + needsPath + " canReach=" + canReach);
```
Comment (reinismu) [inline ElvargServer/src/main/java/com/elvarg/game/content/combat/Combat.java:101]: Remove this

**Review by reinismu — COMMENTED** [review]
* Lots of places commented out that should be removed. 
* Let's remove `@author` stuff especially when someone does such a poor job at it...
* I noticed empty methods that need removing
* Should port over client changes to web client

File: ElvargServer/src/main/java/com/elvarg/net/packet/impl/FollowPlayerPacketListener.java [hunk 3]
```diff
@@ -20,14 +28,47 @@ public void execute(Player player, Packet packet) {
...
+
         Player leader = World.getPlayers().get(otherPlayersIndex);
+
         if (leader == null)
             return;
-        player.setMobileInteraction(leader);
+
         player.setFollowing(leader);
+
+        //System.err.println("following="+leader.getUsername());
```
Comment (tobywisener) [inline ElvargServer/src/main/java/com/elvarg/net/packet/impl/FollowPlayerPacketListener.java:46]: Need to remove

**Review by syuil — APPROVED** [review]
Spent time playtesting the pathing.

Everything seems really smooth on my end. Follow dancing is clean. Running to other players work flawlessly, tested with range (msb + cbows) 
Tested with magics 

Tested with overall running around trees and objects. 

I might be wrong but in 06-07 RS, not OSRS, running around trees could cause the other player to not catch you for one hit or so -- this might have changed though with OSRS. 

Overall, great work here! Really had a good time testing it.

**Comment by tobywisener** [comment]
> Spent time playtesting the pathing.
> 
> Everything seems really smooth on my end. Follow dancing is clean. Running to other players work flawlessly, tested with range (msb + cbows) Tested with magics
> 
> Tested with overall running around trees and objects.
> 
> I might be wrong but in 06-07 RS, not OSRS, running around trees could cause the other player to not catch you for one hit or so -- this might have changed though with OSRS.
> 
> Overall, great work here! Really had a good time testing it.

Thanks very much Syuil. So glad you had a good time testing and I really appreciate your review/test

**Pull request merged**